### ISSUE TITLE
Reproducibility Issue

### ISSUE THREAD

**Issue by MeycL** [kennethorq/SMORE#2]
I am very pleased to see such impressive work being open-sourced. However, I was unable to reproduce the results reported in the paper using the provided hyperparameters. Here are the results I obtained (in MMRec, seed=999):

recall@10: 0.0665 recall@20: 0.1019 ndcg@10: 0.0362 ndcg@20: 0.0453

In particular, the Recall metrics differ significantly from those in the paper. Could the authors provide any suggestions?

**Comment by kennethorq** [comment]
Hi @MeycL, thank you for your interest in our work. We believe the fluctuations in your results may be due to differences in environment configuration. Could you kindly share the details of your environment configuration when training our model? This will help us better understand the issue and provide more accurate assistance/suggestions.

**Comment by kennethorq** [comment]
Hi @MeycL. As there hasn't been any recent activity on this issue, I’ll be closing it for now. If you encounter any new or related issues, please do not hesitate to reach out to us.

**Comment by MeycL** [comment]
Thank you very much for your reply. My configuration is consistent with MMRec, and the hardware environment is Python 3.8, PyTorch 1.11.0+cu113, and NVIDIA 4090 GPU. Although the hardware may differ, the gap shouldn't be this significant, right?

**Comment by Gs0098** [comment]
> Thank you very much for your reply. My configuration is consistent with MMRec, and the hardware environment is Python 3.8, PyTorch 1.11.0+cu113, and NVIDIA 4090 GPU. Although the hardware may differ, the gap shouldn't be this significant, right?非常感谢您的回复。我的配置与 MMRec 一致，硬件环境是 Python 3.8，PyTorch 1.11.0+cu113，以及 NVIDIA 4090 GPU。虽然硬件可能不同，但差距不应该这么大，对吧？

咋样了

**Comment by sun2ot** [comment]
The result I reproduced on the sports dataset is
```
recall@5: 0.0469    recall@10: 0.0720    recall@20: 0.1070    recall@50: 0.1715
ndcg@5: 0.0312    ndcg@10: 0.0393    ndcg@20: 0.0482    ndcg@50: 0.0611
precision@5: 0.0098    precision@10: 0.0076    precision@20: 0.0056    precision@50: 0.0036
map@5: 0.0256    map@10: 0.0289    map@20: 0.0313    map@50: 0.0333   
```

**Comment by Gs0098** [comment]
> The result I reproduced on the sports dataset is我在体育数据集上复现的结果是
> 
> ```
> recall@5: 0.0469    recall@10: 0.0720    recall@20: 0.1070    recall@50: 0.1715
> ndcg@5: 0.0312    ndcg@10: 0.0393    ndcg@20: 0.0482    ndcg@50: 0.0611
> precision@5: 0.0098    precision@10: 0.0076    precision@20: 0.0056    precision@50: 0.0036
> map@5: 0.0256    map@10: 0.0289    map@20: 0.0313    map@50: 0.0333   
> ```

这有点差距大啊0.0 可以交流一下吗

**Comment by sun2ot** [comment]
> 这有点差距大啊0.0 可以交流一下吗

这还是用的reg=1e-5跑的，按照README中写的，sports的正则化权重应该是1e-4...但试了下还不如1e-5

**Comment by Gs0098** [comment]
> > 这有点差距大啊0.0 可以交流一下吗
> 
> 这还是用的reg=1e-5跑的，按照README中写的，sports的正则化权重应该是1e-4...但试了下还不如1e-5

68114346 交流

**Comment by kennethorq** [comment]
> Thank you very much for your reply. My configuration is consistent with MMRec, and the hardware environment is Python 3.8, PyTorch 1.11.0+cu113, and NVIDIA 4090 GPU. Although the hardware may differ, the gap shouldn't be this significant, right?

Hi @MeycL, thank you for your reply. Our environment configuration differs slightly from MMRec. Kindly find our environment settings below for your reference: 

GPU: NVIDIA GeForce RTX 2080 Ti 

Requirements: 
numpy==1.21.5 
pandas==1.3.5 
python==3.7.16 
scipy==1.7.3 
torch== 1.13.1 
pyyaml==6.0 

Based on my experience, even with identical environment settings, training on different GPUs can lead to slight variations. For instance, the approximately 2% difference observed in the Recall@K metrics and the 0.8% difference in the NDCG@K metrics that you have obtained. As long as your experiments are conducted consistently across all baselines, the comparison remains valid. Hope this helps, and please feel free to reach out again if you need any form of assistance.
